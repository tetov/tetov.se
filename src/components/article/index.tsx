export {Article} from "./article"
export {ArticleBody} from "./body"
export {ArticleHeader} from "./header"
export {ArticleListing} from "./listing"
export {ArticlePreview } from "./preview"
export {ArticleTime} from "./time"

export type {ArticleProp} from "./article"
