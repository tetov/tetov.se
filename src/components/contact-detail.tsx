import classNames from "classnames"
import React from "react"

import * as Icons from "../icons"

import type { IconBaseProps } from "react-icons/lib"
import type { Argument as classNamesArgument } from "classnames"

import type { IContactData } from "../types"

type ContactDetailProp = {
  contactData: Partial<IContactData>
  className?: classNamesArgument
  iconProp?: Partial<IconBaseProps>
}

type ContactLinkProp = JSX.IntrinsicElements["a"] & {
  children: React.ReactNode
}

type ContactNotLinkProp = JSX.IntrinsicElements["span"] & {
  children: React.ReactNode
}

const ContactLink = ({ children, ...prop }: ContactLinkProp) => (
  <a rel="me external" {...prop}>
    {children}
  </a>
)

const ContactNotLink = ({ children, ...prop }: ContactNotLinkProp) => (
  <span {...prop}>{children}</span>
)

const ContactDetail = ({
  contactData: { text, url, icon, hcard },
  className,
  iconProp = {},
}: ContactDetailProp) => {
  iconProp["aria-hidden"] = true
  const Icon = icon ? Icons[icon] : undefined

  const prop = {
    children: (
      <>
        {icon && <Icon {...iconProp} />}
        {text || ""}
      </>
    ),
    className: (hcard || className) && classNames(hcard, className),
    href: url || undefined,
  }

  const ContactComponent = url ? ContactLink : ContactNotLink

  return <ContactComponent {...prop} />
}

export default ContactDetail
