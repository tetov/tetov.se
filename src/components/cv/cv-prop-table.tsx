import classNames from "classnames";
import React, { ReactNode } from "react";

export const CVPropTable = ({
  tuples,
}: {
  tuples: [key: string, value: ReactNode][];
}) => {
  const sharedTableCellClassNames = "text-sm px-2 py-1 gray-border";

  return (
    <div className="py-1">
      <table className="table-auto">
        <tbody className="border-t-0 divide-y-0">
          {tuples.map(([key, value]) => (
            <tr key={key} className="gray-border">
              <td
                className={classNames(
                  sharedTableCellClassNames,
                  "font-medium border-r-2 text-right",
                )}
              >
                {key}
              </td>
              <td
                className={classNames(sharedTableCellClassNames, "font-light")}
              >
                {value}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
