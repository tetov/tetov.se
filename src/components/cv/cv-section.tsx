import classNames from "classnames";
import React, { PropsWithChildren } from "react";
import { LEFT_COL_WIDTH } from ".";

export const CVSection = ({
  title,
  children,
}: PropsWithChildren<{ title: string }>) => {
  // check if null or array len 0
  if (!children || (Array.isArray(children) && children.length === 0)) {
    return null;
  }
  return (
    <section>
      <div className="mt-8 mb-2 flex-row flex [page-break-inside:avoid]">
        {/* width set up like CVEntry */}
        <div className={classNames("flex", LEFT_COL_WIDTH)} />
        <div className="flex ml-2 w-full max-w-prose">
          <h2 className="text-2xl">{title}</h2>
        </div>
      </div>
      <hr className="gray-border border-y-2" />
      <div className="border-collapse">{children}</div>
    </section>
  );
};
