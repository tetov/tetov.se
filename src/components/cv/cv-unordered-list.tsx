import React, { PropsWithChildren } from "react";

export const CVUnorderedList = ({ children }: PropsWithChildren) => (
  <ul className="list-inside list-disc marker:text-purple print:marker:text-light-text">
    {children}
  </ul>
);
