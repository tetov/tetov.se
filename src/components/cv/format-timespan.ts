const cvFormatTimespan = ({
  startDate,
  endDate,
}: {
  startDate?: string;
  endDate?: string;
}): string => {
  // if both are present and same
  if (startDate && endDate && startDate == endDate) {
    return startDate;
  }
  // if both are present and unique
  if (startDate && endDate) {
    return `${startDate} â€“ ${endDate}`;
  }

  // if none are present
  if (!startDate && !endDate) {
    return "";
  }

  // one is missing
  // if endDate is present, return it
  // if startDate is present, return it with a dash after
  return endDate ?? `${startDate} â€“`;
};

export default cvFormatTimespan;
