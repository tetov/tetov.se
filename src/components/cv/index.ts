export { default as cvFormatTimespan } from "./format-timespan";

export { CVSection } from "./cv-section";
export { CVEntry, CVEntryBody } from "./cv-entry";
export { CVPropTable } from "./cv-prop-table";
export { CVUnorderedList } from "./cv-unordered-list";

export const LEFT_COL_WIDTH = "w-1/4";
