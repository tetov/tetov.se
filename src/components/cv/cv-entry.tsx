import classNames from "classnames";
import React, { PropsWithChildren, ReactNode } from "react";
import { cvFormatTimespan } from "src/components/cv";
import { LEFT_COL_WIDTH } from "./";

type CVEntryProp = {
  startDate?: string;
  endDate?: string;
  heading?: React.ReactNode;
  url?: string;
};
export const CVEntry = ({
  startDate,
  endDate,
  heading,
  children,
}: PropsWithChildren<CVEntryProp>) => (
  <div className="mt-6 mb-4 flex-row flex">
    <div
      className={classNames(
        "flex text-base tracking-tight pt-1 mr-2",
        LEFT_COL_WIDTH,
      )}
    >
      {cvFormatTimespan({ startDate, endDate })}
    </div>
    {(heading && (
      <div className="flex w-full flex-col max-w-prose">
        <div className="flex">
          <h3 className="mb-1 font-bold tracking-widest text-md">{heading}</h3>
        </div>
        <div className="flex flex-col">{children}</div>
      </div>
    )) || <div className="w-full max-w-prose">{children}</div>}
  </div>
);

export const CVEntryBody = ({
  children,
  description,
}: PropsWithChildren<{ description: ReactNode }>) => (
  <>
    <div className="flex max-w-prose font-light whitespace-pre-line">
      {description}
    </div>
    <div className="pt-2 flex min-w-full">{children}</div>
  </>
);
