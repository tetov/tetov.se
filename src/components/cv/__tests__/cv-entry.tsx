import React from "react";
import renderer from "react-test-renderer";
import { CVEntry, CVEntryBody } from "src/components/cv";

describe("CVEntry", () => {
  it("renders correctly without props", () => {
    const tree = renderer.create(<CVEntry />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with all fields", () => {
    const tree = renderer
      .create(
        <CVEntry
          startDate="2018-10-02"
          endDate="2020-10-02"
          key="key"
          heading="Farmhand in Stardew Valley"
          url="https://stardewvalley.net/"
        >
          Hard work!
        </CVEntry>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly without enddate", () => {
    const tree = renderer
      .create(
        <CVEntry
          startDate="2018-10-02"
          key="key"
          heading="Farmhand in Stardew Valley"
          url="https://stardewvalley.net/"
        >
          Hard work!
        </CVEntry>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("CVEntryBody", () => {
  it("renders correctly with text child", () => {
    const tree = renderer
      .create(<CVEntryBody description="Test">Test</CVEntryBody>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with html child", () => {
    const tree = renderer
      .create(
        <CVEntryBody description="Test">
          <ul>
            <li>Test 1</li>
            <li>Test 2</li>
          </ul>
        </CVEntryBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with html description", () => {
    const tree = renderer
      .create(
        <CVEntryBody description={<b>test</b>}>
          <ul>
            <li>Test 1</li>
            <li>Test 2</li>
          </ul>
        </CVEntryBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Combined CVEntry & CVEntryBody", () => {
  it("renders correctly with text child", () => {
    const tree = renderer
      .create(
        <CVEntry>
          <CVEntryBody description="Test">Test</CVEntryBody>
        </CVEntry>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with CVEntry props", () => {
    const tree = renderer
      .create(
        <CVEntry
          startDate="2018-10-02"
          endDate="2020-10-02"
          key="key"
          heading="Farmhand in Stardew Valley"
          url="https://stardewvalley.net/"
        >
          <CVEntryBody description="Test">Test</CVEntryBody>
        </CVEntry>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
