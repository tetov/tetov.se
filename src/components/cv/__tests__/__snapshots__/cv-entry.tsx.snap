// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEntry renders correctly with all fields 1`] = `
<div
  className="mt-6 mb-4 flex-row flex"
>
  <div
    className="flex text-base tracking-tight pt-1 mr-2 w-1/4"
  >
    2018-10-02 – 2020-10-02
  </div>
  <div
    className="flex w-full flex-col max-w-prose"
  >
    <div
      className="flex"
    >
      <h3
        className="mb-1 font-bold tracking-widest text-md"
      >
        Farmhand in Stardew Valley
      </h3>
    </div>
    <div
      className="flex flex-col"
    >
      Hard work!
    </div>
  </div>
</div>
`;

exports[`CVEntry renders correctly without enddate 1`] = `
<div
  className="mt-6 mb-4 flex-row flex"
>
  <div
    className="flex text-base tracking-tight pt-1 mr-2 w-1/4"
  >
    2018-10-02 –
  </div>
  <div
    className="flex w-full flex-col max-w-prose"
  >
    <div
      className="flex"
    >
      <h3
        className="mb-1 font-bold tracking-widest text-md"
      >
        Farmhand in Stardew Valley
      </h3>
    </div>
    <div
      className="flex flex-col"
    >
      Hard work!
    </div>
  </div>
</div>
`;

exports[`CVEntry renders correctly without props 1`] = `
<div
  className="mt-6 mb-4 flex-row flex"
>
  <div
    className="flex text-base tracking-tight pt-1 mr-2 w-1/4"
  />
  <div
    className="w-full max-w-prose"
  />
</div>
`;

exports[`CVEntryBody renders correctly with html child 1`] = `
[
  <div
    className="flex max-w-prose font-light whitespace-pre-line"
  >
    Test
  </div>,
  <div
    className="pt-2 flex min-w-full"
  >
    <ul>
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
    </ul>
  </div>,
]
`;

exports[`CVEntryBody renders correctly with html description 1`] = `
[
  <div
    className="flex max-w-prose font-light whitespace-pre-line"
  >
    <b>
      test
    </b>
  </div>,
  <div
    className="pt-2 flex min-w-full"
  >
    <ul>
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
    </ul>
  </div>,
]
`;

exports[`CVEntryBody renders correctly with text child 1`] = `
[
  <div
    className="flex max-w-prose font-light whitespace-pre-line"
  >
    Test
  </div>,
  <div
    className="pt-2 flex min-w-full"
  >
    Test
  </div>,
]
`;

exports[`Combined CVEntry & CVEntryBody renders correctly with CVEntry props 1`] = `
<div
  className="mt-6 mb-4 flex-row flex"
>
  <div
    className="flex text-base tracking-tight pt-1 mr-2 w-1/4"
  >
    2018-10-02 – 2020-10-02
  </div>
  <div
    className="flex w-full flex-col max-w-prose"
  >
    <div
      className="flex"
    >
      <h3
        className="mb-1 font-bold tracking-widest text-md"
      >
        Farmhand in Stardew Valley
      </h3>
    </div>
    <div
      className="flex flex-col"
    >
      <div
        className="flex max-w-prose font-light whitespace-pre-line"
      >
        Test
      </div>
      <div
        className="pt-2 flex min-w-full"
      >
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`Combined CVEntry & CVEntryBody renders correctly with text child 1`] = `
<div
  className="mt-6 mb-4 flex-row flex"
>
  <div
    className="flex text-base tracking-tight pt-1 mr-2 w-1/4"
  />
  <div
    className="w-full max-w-prose"
  >
    <div
      className="flex max-w-prose font-light whitespace-pre-line"
    >
      Test
    </div>
    <div
      className="pt-2 flex min-w-full"
    >
      Test
    </div>
  </div>
</div>
`;
